File: tugas-no 1.php
class BangunDatar {
    public function hitungLuas() {
        return "Hitung luas bangun datar.";
    }

    public function hitungKeliling() {
        return "Hitung keliling bangun datar.";
    }
}

class Persegi extends BangunDatar {
    public $sisi;

    public function __construct($sisi) {
        $this->sisi = $sisi;
    }

    public function hitungLuas() {
        return "Luas persegi: " . ($this->sisi * $this->sisi);
    }

    public function hitungKeliling() {
        return "Keliling persegi: " . (4 * $this->sisi);
    }
}

class PersegiPanjang extends BangunDatar {
    public $panjang;
    public $lebar;

    public function __construct($panjang, $lebar) {
        $this->panjang = $panjang;
        $this->lebar = $lebar;
    }

    public function hitungLuas() {
        return "Luas persegi panjang: " . ($this->panjang * $this->lebar);
    }

    public function hitungKeliling() {
        return "Keliling persegi panjang: " . (2 * ($this->panjang + $this->lebar));
    }
}

class Segitiga extends BangunDatar {
    public $alas;
    public $tinggi;

    public function __construct($alas, $tinggi) {
        $this->alas = $alas;
        $this->tinggi = $tinggi;
    }

    public function hitungLuas() {
        return "Luas segitiga: " . (0.5 * $this->alas * $this->tinggi);
    }

    public function hitungKeliling() {
        return "Hitung keliling segitiga.";
    }
}

class Lingkaran extends BangunDatar {
    public $jariJari;

    public function __construct($jariJari) {
        $this->jariJari = $jariJari;
    }

    public function hitungLuas() {
        return "Luas lingkaran: " . (3.14 * $this->jariJari * $this->jariJari);
    }

    public function hitungKeliling() {
        return "Keliling lingkaran: " . (2 * 3.14 * $this->jariJari);
    }
}

// Contoh penggunaan
$persegi = new Persegi(5);
echo $persegi->hitungLuas() . "<br>";
echo $persegi->hitungKeliling() . "<br>";

$persegiPanjang = new PersegiPanjang(4, 6);
echo $persegiPanjang->hitungLuas() . "<br>";
echo $persegiPanjang->hitungKeliling() . "<br>";

$segitiga = new Segitiga(3, 4);
echo $segitiga->hitungLuas() . "<br>";
echo $segitiga->hitungKeliling() . "<br>";

$lingkaran = new Lingkaran(7);
echo $lingkaran->hitungLuas() . "<br>";
echo $lingkaran->hitungKeliling() . "<br>";

File: tugas no 1.cpp
#include <iostream>

using namespace std;

class BangunDatar {
public:
    virtual double hitungLuas() {
        return 0;
    }

    virtual double hitungKeliling() {
        return 0;
    }
};

class Persegi : public BangunDatar {
private:
    double sisi;

public:
    Persegi(double s) : sisi(s) {}

    double hitungLuas() override {
        return sisi * sisi;
    }

    double hitungKeliling() override {
        return 4 * sisi;
    }
};

class PersegiPanjang : public BangunDatar {
private:
    double panjang;
    double lebar;

public:
    PersegiPanjang(double p, double l) : panjang(p), lebar(l) {}

    double hitungLuas() override {
        return panjang * lebar;
    }

    double hitungKeliling() override {
        return 2 * (panjang + lebar);
    }
};

class Segitiga : public BangunDatar {
private:
    double alas;
    double tinggi;

public:
    Segitiga(double a, double t) : alas(a), tinggi(t) {}

    double hitungLuas() override {
        return 0.5 * alas * tinggi;
    }

    double hitungKeliling() override {
        // Hitung keliling segitiga tidak diimplementasikan dalam contoh ini.
        return 0;
    }
};

class Lingkaran : public BangunDatar {
private:
    double jariJari;

public:
    Lingkaran(double r) : jariJari(r) {}

    double hitungLuas() override {
        return 3.14 * jariJari * jariJari;
    }

    double hitungKeliling() override {
        return 2 * 3.14 * jariJari;
    }
};

int main() {
    Persegi persegi(5);
    cout << "Luas Persegi: " << persegi.hitungLuas() << endl;
    cout << "Keliling Persegi: " << persegi.hitungKeliling() << endl;

    PersegiPanjang persegiPanjang(4, 6);
    cout << "Luas Persegi Panjang: " << persegiPanjang.hitungLuas() << endl;
    cout << "Keliling Persegi Panjang: " << persegiPanjang.hitungKeliling() << endl;

    Segitiga segitiga(3, 4);
    cout << "Luas Segitiga: " << segitiga.hitungLuas() << endl;

    Lingkaran lingkaran(7);
    cout << "Luas Lingkaran: " << lingkaran.hitungLuas() << endl;
    cout << "Keliling Lingkaran: " << lingkaran.hitungKeliling() << endl;

    return 0;
}
